code/modules/mob/living/carbon/inventory.dm

```diff
@@ -141,3 +141,61 @@
 /mob/living/carbon/proc/get_holding_bodypart_of_item(obj/item/I)
 	var/index = get_held_index_of_item(I)
 	return index && hand_bodyparts[index]
+
+/**
+ * Proc called when giving an item to another player
+ *
+ * This handles creating an alert and adding an overlay to it
+ */
+
+/mob/living/carbon/proc/give()
+	var/alert = null
+	var/obj/item/receiving = get_active_held_item()
+	if(!receiving)
+		to_chat(src, "<span class='warning'>You're not holding anything to give!</span>")
+		return
+
+	for(var/mob/living/carbon/C in orange(1, src)) //Fixed that, now it shouldn't be able to give benos stunbatons and IDs
+		if(!CanReach(C) || !C.can_hold_items())
+			continue
+
+		var/atom/movable/screen/alert/give/G = C.throw_alert("[src]", /atom/movable/screen/alert/give)
+		if(!G)
+			to_chat(src, "<span class='warning'>There is nobody nearby to give [receiving]!</span>")
+			continue
+
+		G.setup(C, src, receiving)
+
+		if(!alert)
```

Comment:
alert will always be null when this is checked, it doesn't look like setting it to true before the proc ends does do anything either

Reply:
It's there so that only a single alert is produced even if the loop finds multiple targets.

Reply:
You can test it by having two nearby targets to give the item to, alert will be 1 and thus only send the visible_message once instead of twice, but both people will receive the alert and can receive the item.